web_pwa_oca/static/src/js/pwa_install.js

```diff
@@ -1,45 +1,38 @@
-odoo.define('web_pwa_oca.systray.install', function (require) {
-"use strict";
+/* Copyright 2020 Lorenzo Battistini @ TAKOBI
+ * License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). */
 
-var core = require('web.core');
-var session = require('web.session');
-var UserMenu = require('web.UserMenu');
+odoo.define('web_pwa_oca.systray.install', function(require) {
+    "use strict";
 
-if ('serviceWorker' in navigator) {
-  window.addEventListener('load', function () {
-    navigator.serviceWorker.register('/service-worker.js')
-        .then(function (reg) {
-          console.log('Service worker registered.', reg);
-        });
-  });
-}
+    var UserMenu = require('web.UserMenu');
 
-var deferredInstallPrompt = null;
+    var deferredInstallPrompt = null;
 
-UserMenu.include({
-    start: function () {
-        window.addEventListener('beforeinstallprompt', this.saveBeforeInstallPromptEvent);
-        return this._super.apply(this, arguments);
-    },
-    saveBeforeInstallPromptEvent: function(evt) {
-        deferredInstallPrompt = evt;
-        this.$.find('#pwa_install_button')[0].removeAttribute('hidden');
-    },
-    _onMenuInstallpwa: function () {
-        deferredInstallPrompt.prompt();
-        // Hide the install button, it can't be called twice.
-        this.el.setAttribute('hidden', true);
-        // Log user response to prompt.
-        deferredInstallPrompt.userChoice
-            .then(function (choice) {
-              if (choice.outcome === 'accepted') {
-                console.log('User accepted the A2HS prompt', choice);
-              } else {
-                console.log('User dismissed the A2HS prompt', choice);
-              }
-              deferredInstallPrompt = null;
-            });
-    },
-});
+
+    UserMenu.include({
+        start: function() {
+            window.addEventListener('beforeinstallprompt', this.saveBeforeInstallPromptEvent);
```

Comment:
For 10.0, we ran into a condition where the 'beforeinstallprompt' handler is not fired, because this event listener is attached too late: the UserMenu's "start" function gets executed far after "serviceWorker.register", so the event can be missed, and then the 'Install PWA' button is not shown.

[Someone else](https://stackoverflow.com/q/61133389) ran into this as well, then solved it by putting the two closer together, which is also what worked for me [in the 10.0 backport](https://github.com/OCA/web/pull/1745/files#diff-ef9b369bb7e8416ad7be0e69b3247823580cfbbc86bfc72a3f418e8363bb80e6R17).

In your code, I see that you've moved "serviceWorker.register" somewhere else, but the two are still not in the same place, so I don't know in which order they are called in your case... but as a warning

Reply:
Thanks for the advice. I see that custom pwa installation flow has some problems.
If you refresh the page you still see the install option? I need restart the browser to see it again :/

Reply:
For us it was also erratic before but now it always comes up, also after browser refresh. You'll need to do something like [this](https://github.com/OCA/web/pull/1745/commits/7bf04cd9bf4b5ad3a408f07cfc771e199412c6fe).

Reply:
I have tried your PR for 10.0 and I also have the same issues ... I don't know if it's a problem because I'm in localhost.

Reply:
Actually yes, that can be an issue:

![image](https://user-images.githubusercontent.com/1466356/100987337-b0829080-354e-11eb-9331-9a84bf4ba354.png)

Reply:
You probably don't have HTTPS on your localhost.